tests/api/controllers/center_of_mass_and_contact_points/center_of_mass_and_contact_points.c

```diff
@@ -52,7 +52,20 @@ int main(int argc, char **argv) {
   epsilon = 0.0;
   ts_assert_boolean_equal(epsilon <= THRESHOLD, "The contact points positions do not match the reference positions.");
 
-  bool stable = wb_supervisor_node_get_static_balance(node);
+  // contact points (descendants included) checks
+  number_of_contact_points = wb_supervisor_node_get_number_of_contact_points(node, true);
+  ts_assert_boolean_equal(number_of_contact_points == REFERENCE_NUMBER_OF_CONTACT_POINTS + ADDITIONAL_CONTACT_POINTS_NUMBER,
+                          "Wrong number of contact points when including descendants.");
+
+  ts_assert_boolean_equal(wb_supervisor_node_get_contact_point_node(node, 0) == node,
+                          "First node contact point should belong to the main node.");
+
+  ts_assert_boolean_equal(wb_supervisor_node_get_contact_point_node(
+                            node, REFERENCE_NUMBER_OF_CONTACT_POINTS + ADDITIONAL_CONTACT_POINTS_NUMBER - 1) != node,
+                          "First node contact point should not belong to the main node.");
+
```

Comment:
The test should go a bit further and test that the last contact point actually belongs to the SECONDARY_BODY solid.